Tests/NimbleTests/XCTestManifests.swift

```diff
@@ -344,6 +344,27 @@ extension UserDescriptionTest {
     ]
 }
 
+#if swift(>=5)
+extension BeSuccessTest {
```

Comment:
This file is generated by `swift test --generate-linuxmain` so please don't edit manually.

Reply:
Ok. Fixed that.

Reply:
@ikesyo I think it's never worked automatically. Because in order to ```swift test --generate-linuxmain``` to work the code of test target must be compiled, and it can't be compiled because new test class doesn't conform to XCTestCaseProviderStatic through XCTestCaseProvider because ```__allTests``` property haven't been generated yet. So the manual addition of extension into XCTestManifests.swift for new test file with empty ```__allTests``` array is necessary. In fact if you delete or rename any existing test you would have to manually delete or rename it in XCTestManifests.swift. 
Secondly ```swift test``` generates slightly different code in swift 5. Specifically it generates all tests property with test class name suffix ```static let __allTests__BeNilTest``` so it breaks all tests because all test class become nonconforming to XCTestCaseProviderStatic protocol. So it will need manual editing of XCTestManifests or modifying all test class and removing conformation to XCTestCaseProvider, effectively removing detecting of missing test cases.
If we will go the second way, removing XCTestCaseProvider, we can still automate generation of XCTestManifests via running ```swift test --generate-linuxmain``` before every test and excluding XCTestManifests.swift from VCS. I've done that in [here](https://github.com/xzeror/Nimble/tree/result_matchers_and_swift_test_autogeneration). If it should be done in separate request tell me please.
So I think that most straight way is to merge this PR with manually edited XCTestManifests. And then move to full XCTestManifests autogeneration.

Reply:
Thanks for letting me know that. I didn't updated the manifest with Swift 5 yet, so didn't know that. But

> If we will go the second way, removing XCTestCaseProvider, we can still automate generation of XCTestManifests via running swift test --generate-linuxmain before every test and excluding XCTestManifests.swift from VCS. I've done that in here.

This should not work because

- `swift test --generate-linuxmain` only works on macOS (thanks to Objective-C runtime)
- and LinuxMain is for Linux, not for SwiftPM on maxOS so we can't use the file without commiting it
  - If `swift test --generate-linuxmain` works on Linux, that means that automatic discovery works so there is no need to prepare the manifest file. That is because LinuxMain exists

Reply:
TL;DR

- Generate LinuxMain on macOS
- Use LinuxMain (which is generated on macOS) on Linux

That is the workflow.

Reply:
@ikesyo ok. Got it. So should I regenerate XCTestManifests using Swift 5 toolchain and remove XCTestCaseProvider stuff in this PR (I guess this won't be able to be build on Swift 4.2.1) or try to resolve build issues with manually edited current XCTestManifests and move XCTestManifests to Swift 5 in separate PR?